### ISSUE TITLE
Cluster version information

### ISSUE THREAD

**Issue by MikeGoldsmith** [couchbaselabs/sdk-rfcs#71]
It would be desirable for the client to be able to know what the cluster version it is connect to is. This would be useful for determining if a particular feature is available or not. Also, consuming applications, projects could have additional requirements.

Cluster version information is available using the `http://localhost:8091/pools/default`. An example for parsing the response can be seen [here](http://src.couchbase.org/source/xref/trunk/ns_server/priv/public/ui/app-classic/components/mn_pool_default.js#10).

Considerations:
- Minimum compatibility cluster version
- Per node version
- Caching behaviour
- Version information (major, minor, patch, build and edition)

**Comment by daschl** [comment]
I think thats a great idea, and java has this already to varying degrees (i.e. we use it for test setup internally to figure out which tests to run). Other parts are exposed on the management APIs for retrieval purposes.

What could make sense is to collect the status quo for our SDKs and see what we currently have (and in parallel of course define where we want to be :)) 👍

**Comment by MikeGoldsmith** [comment]
Yeah, agree we should use what the SDKs already have implemented and see if it fits all of use-cases. I'll make a start on the RFC to find out what we already have.
